src/crypto/generichash/mod.rs

```diff
@@ -155,7 +155,8 @@ mod test {
         loop {
             let msg = {
                 line.clear();
-                if let Err(_) = r.read_line(&mut line) {
+                r.read_line(&mut line).unwrap();
+                if line.is_empty() {
```

Comment:
Are these semantically equivalent?

Reply:
I took the same code from https://github.com/sodiumoxide/sodiumoxide/pull/337/files#diff-c75fa680a5ac2678ede6c3c657ef37b0L71 and https://github.com/sodiumoxide/sodiumoxide/pull/337/files#diff-e56ac0ce53049e9b4c3da2aaac1f530aR56 .

Reply:
I took the same code from https://github.com/sodiumoxide/sodiumoxide/pull/337/files#diff-c75fa680a5ac2678ede6c3c657ef37b0L71 and https://github.com/sodiumoxide/sodiumoxide/pull/337/files#diff-e56ac0ce53049e9b4c3da2aaac1f530aR56 .

Reply:
Oh you are right. There are empty lines in testvectors/blake2b-kat.txt. I tested them, if fileline is empty, then line="\n", so line.is_empty() only at the EOF.

> Are these semantically equivalent?

yes